test/integration/integration.test.ts

```diff
@@ -109,485 +65,222 @@ describe('Integration tests', (): void => {
-
-      expect(response.code).toBe('200')
-      assert(response.quote)
-
-      quote = response.quote
-    })
-    test('Create Outgoing Payment', async (): Promise<void> => {
-      const handleWebhookEventSpy = jest.spyOn(
-        c9.integrationServer.webhookEventHandler,
-        'handleWebhookEvent'
+    test('Peer to Peer - Cross Currency', async (): Promise<void> => {
+      const {
+        createReceiver,
+        createQuote,
+        createOutgoingPayment,
+        getOutgoingPayment
+      } = testActions.admin
+
+      const senderWalletAddress = await c9.accounts.getByWalletAddressUrl(
+        'https://host.docker.internal:3100/accounts/gfranklin'
       )
-
-      const response = await c9.adminClient.createOutgoingPayment({
-        walletAddressId: gfranklinWalletAddressId,
-        quoteId: quote.id
-      })
-
-      expect(response.code).toBe('200')
-      assert(response.payment)
-
-      outgoingPayment = response.payment
-
-      await pollCondition(
-        () => {
-          return (
-            handleWebhookEventSpy.mock.calls.some(
-              (call) =>
-                call[0]?.type === WebhookEventType.OutgoingPaymentCreated
-            ) &&
-            handleWebhookEventSpy.mock.calls.some(
-              (call) =>
-                call[0]?.type === WebhookEventType.OutgoingPaymentCompleted
-            )
-          )
+      assert(senderWalletAddress?.walletAddressID)
+      const senderWalletAddressId = senderWalletAddress.walletAddressID
+      const value = '500'
+      const createReceiverInput = {
+        metadata: {
+          description: 'cross-currency'
         },
-        5,
-        0.5
-      )
+        incomingAmount: {
+          assetCode: 'EUR',
+          assetScale: 2,
+          value: value as unknown as bigint
+        },
+        walletAddressUrl: 'https://host.docker.internal:4100/accounts/lars'
+      }
 
-      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          type: WebhookEventType.OutgoingPaymentCreated,
-          data: expect.any(Object)
-        })
-      )
-      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          type: WebhookEventType.OutgoingPaymentCompleted,
-          data: expect.any(Object)
-        })
-      )
-    })
-    test('Get Outgoing Payment', async (): Promise<void> => {
-      const payment = await c9.adminClient.getOutgoingPayment(
-        outgoingPayment.id
+      const receiver = await createReceiver(createReceiverInput)
+      const quote = await createQuote(senderWalletAddressId, receiver)
+      const outgoingPayment = await createOutgoingPayment(
+        senderWalletAddressId,
+        quote
       )
-      expect(payment.state).toBe(OutgoingPaymentState.Completed)
-      expect(payment.receiveAmount.value).toBe(amountValueToSend)
-      expect(payment.sentAmount.value).toBe(amountValueToSend)
+      await getOutgoingPayment(outgoingPayment.id, value)
```

Comment:
> want to assume a certain conversion rate/fee structure was used

Since we have access to the config file, would we be able to get the rates from there and confirm the amounts that way?

Reply:
I think there are fees too, so we'd need to get that from the seed as well. And then duplicate the non-trivial backend logic and live with the fact that it could change, or extract that out and use it here. What do you think? Seem like not great options IMO. I suppose maybe if really want to verify those amounts I can just hardcode what they currently are and state the assumptions in a comment (fee, rates, starting amount)? 

I guess im not sure, but perhaps those details aren't really that critical in this test? We're testing (or should be) those calculations and whether or not they're returned in the backend already. I think the important part of this test is that we can get the payment and see that its complete.

Reply:
Maybe checking that the conversion went in the right direction would be good enough? Like if we converted between currencies with a 2:1 ratio that the amount went down as opposed to up?